odrive-motor/src/odriveCAN.py

```diff
@@ -0,0 +1,290 @@
+        await self.try_send(msg)
+        await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
+
+        if revolutions == 0.0:
+            msg = db.get_message_by_name('Set_Controller_Mode')
+            data = msg.encode({'Control_Mode': 0x02, 'Input_Mode': 0x01})
+            msg = can.Message(arbitration_id=0x00B | self.nodeID << 5, is_extended_id=False, data=data)
+            await self.try_send(msg)
+            
+            msg = db.get_message_by_name('Set_Input_Vel')
+            data = msg.encode({'Input_Vel': rps, 'Input_Torque_FF': 0})
+            msg = can.Message(arbitration_id=0x00D | self.nodeID << 5, is_extended_id=False, data=data)
+            await self.try_send(msg)
+
+        else: 
+            msg = db.get_message_by_name('Set_Controller_Mode')
+            data = msg.encode({'Control_Mode': 0x03, 'Input_Mode': 0x05})
+            msg = can.Message(arbitration_id=0x00B | self.nodeID << 5, is_extended_id=False, data=data)
+            await self.try_send(msg)
+
+            msg = db.get_message_by_name('Set_Limits')
+            data = msg.encode({'Velocity_Limit': 10.0, 'Current_Limit': 40.0})
+            msg = can.Message(arbitration_id=0x00F | self.nodeID << 5, is_extended_id=False, data=data)
+            await self.try_send(msg)
+
+            current_position = await self.get_position()
+            if rpm > 0.0:
+                msg = db.get_message_by_name('Set_Input_Pos')
+                goal_position = current_position+revolutions+self.offset
+                data = msg.encode({'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
+                msg = can.Message(arbitration_id=0x00C | self.nodeID << 5, is_extended_id=False, data=data)
+                await self.try_send(msg)
+            else:
+                msg = db.get_message_by_name('Set_Input_Pos')
+                goal_position = current_position-revolutions+self.offset
+                data = msg.encode({'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
+                msg = can.Message(arbitration_id=0x00C | self.nodeID << 5, is_extended_id=False, data=data)
+                await self.try_send(msg)
+
+            await self.wait_and_set_to_idle(goal_position)
+
+    async def go_to(self, rpm: float, revolutions: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        current_position = await self.get_position()
+        revolutions = revolutions - current_position
+        await self.go_for(rpm, revolutions)
+
+    async def reset_zero_position(self, offset: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        position = await self.get_position()
+        self.offset += position
+
+    async def get_position(self, extra: Optional[Dict[str, Any]] = None, **kwargs) -> float:
+        msg = db.get_message_by_name('Get_Encoder_Count')
+        data1 = msg.encode({'Shadow_Count': 0, 'Count_in_CPR': 4000})
```

Comment:
This code is often repeated, can we make it a function so it's easier to maintain?